challengeMk2/ViewModels/PuzzleViewModel.cs

```diff
@@ -0,0 +1,265 @@
+            var apiBaseAddress = Device.RuntimePlatform == Device.Android ? "https://10.0.2.2:5001" : "https://localhost:5001";    // Android Emulator on Windows 10
+
+            string apiRoute;
+            HttpResponseMessage apiResponse;
+            string responseContent;
+
+            var isPuzzleRunning = true;
+            while (isPuzzleRunning)
+            {
+                apiRoute = $"{apiBaseAddress}/api/TheNumber/{userTry}";
+                apiResponse = await client.GetAsync(apiRoute);
+                responseContent = await apiResponse.Content.ReadAsStringAsync();
+
+                switch (apiResponse.StatusCode)
+                {
+                    case HttpStatusCode.OK:
+                        currentResult = GetResult(HttpStatusCode.OK, responseContent, userTry);
+                        CompareTry(currentResult, ref min, ref max, ref userTry);
+                        break;
+
+                    case HttpStatusCode.Accepted:
+                        currentResult = GetResult(HttpStatusCode.Accepted, responseContent, userTry);
+                        isPuzzleRunning = false;
+                        CanRunPuzzle = true;
+                        ButtonText = "Retry";
+                        UpdateStatistics(currentResult.TryNumber);
+                        break;
+
+                    case HttpStatusCode.ResetContent:
+                        currentResult = CreateResult(HttpStatusCode.ResetContent, 20, apiResponse.ReasonPhrase, userTry);
+                        isPuzzleRunning = false;
+                        CanRunPuzzle = true;
+                        ButtonText = "You should modify something...";
+                        break;
+
+                    case HttpStatusCode.InternalServerError:
+                        currentResult = CreateResult(HttpStatusCode.InternalServerError, 0, apiResponse.ReasonPhrase, userTry);
+                        isPuzzleRunning = false;
+                        break;
+                }
+
+                Tries.Insert(0, currentResult);
+
+                await Task.Delay(1000);
+            }
+        }
+
+
+
+        int GetMiddle(int min, int max)
+        {
+            return min + ((max - min) / 2);
+        }
+
+        TryResult GetResult(HttpStatusCode status, string apiResponseContent, int userTry)
+        {
+            var apiResult = JsonConvert.DeserializeObject<TryResult>(apiResponseContent);
+            apiResult.UserTry = userTry;
+            apiResult.Status = status;
+
+            return apiResult;
+        }
+
+        TryResult CreateResult(HttpStatusCode status, int tryNumber, string result, int userTry)
+        {
+            var apiResult = new TryResult
+            {
+                TryNumber = tryNumber,
+                Result = result,
+                UserTry = userTry,
+                Status = status
+            };
+
+            return apiResult;
+        }

```

Comment:
Why is this method here ? (Encapsulation !)

Reply:
Yes, the model should not have to call the api and then work on response to retrieve a valide result.
To change that, i add a puzzle service,  who will be in charge to communicate with the api and return a valide result for the view Model. 
The view Model just displays the result and choose the number for the next try before send it to Puzzle service.

Reply:
I will create a new branch for the animations, to keep it clean ;D